lib/tasks/one_time.rake

```diff
@@ -8,4 +8,17 @@ namespace :one_time do
   task assign_ui_and_gem_signin: :environment do
     User.where(mfa_level: :disabled).where(id: WebauthnCredential.select(:user_id)).update_all(mfa_level: :ui_and_gem_signin)
   end
+
+  desc "Hash mfa recovery codes"
+  task hash_recovery_codes: :environment do
+    User.where(hashed_mfa_recovery_codes: nil).where.not(mfa_recovery_codes: nil).find_each do |user|
+      user.hashed_mfa_recovery_codes = user.mfa_recovery_codes.map { |code| BCrypt::Password.create(code) }
+      user.save!(validate: false)
+    end
+  end
+
+  desc "Remove plain text recovery codes"
+  task remove_plain_text_recovery_codes: :environment do
+    User.where.not(mfa_recovery_codes: nil).update_all(mfa_recovery_codes: [])
+  end
```

Comment:
Is the thinking to remove application references of `mfa_recovery_codes` first, then run this task? If this is run before, `mfa_recovery_codes` will be later populated if someone generates codes. 

If so, what is the difference between setting the column to `[]` vs just deleting the column?

Reply:
true -- i will nuke that task